src/test/java/org/assertj/core/internal/files/Files_assertIsDirectoryRecursivelyContaining_SyntaxAndPattern_Test.java

```diff
@@ -0,0 +1,138 @@
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = { "regex:.+\\.data", "regex:.+foobar.*", "regex:.+root.+foo.*" })
+  void should_pass_if_actual_contains_some_files_matching_the_given_pathMatcherPattern(String pattern) {
+    // GIVEN
+    createDefaultFixturePaths();
+    // WHEN-THEN
+
+    files.assertIsDirectoryRecursivelyContaining(INFO, tempDirAsFile, pattern);
+  }
+
+  private void createDefaultFixturePaths() {
+    // @format:off
+    // The layout :
+    //  /root/
+    //       /root/foo/
+    //       /root/foo/foobar
+    //       /root/foo/foobar/foobar1.data
+    //       /root/foo/foobar/foobar2.json
+    //       /root/foo2.data
+    // @format:on
+    Path rootDir = createDirectoryWithDefaultParent("root", "foo2.data");
+    Path fooDir = createDirectory(rootDir, "foo");
+    createDirectory(fooDir, "foobar", "foobar1.data", "foobar2.json");
+  }
+
+  @Test
+  void should_pass_if_all_actual_files_matching_the_given_pathMatcherPattern() {
+    Path fooDir = createDirectory(tempDir, "foo", "foo2.data");
+    createDirectory(fooDir, "foo3");
+    files.assertIsDirectoryRecursivelyContaining(INFO, tempDirAsFile, "regex:.*foo.*|.*tmp");
+  }
+
+  @Test
+  void should_fail_if_actual_does_not_exist() {
+    // GIVEN
+    File notExistingFile = new File("foo/bar/doesnt-exist-file");
+    // WHEN
+    expectAssertionError(() -> files.assertIsDirectoryRecursivelyContaining(INFO, notExistingFile,
+                                                                            TXT_EXTENSION_PATTERN));
+    // THEN
+    verify(failures).failure(INFO, shouldBeDirectory(notExistingFile));
+  }
+
+  @Test
+  void should_fail_if_actual_exists_but_is_not_a_directory() throws IOException {
+    // GIVEN
+    File existingFile = java.nio.file.Files.createFile(tempDir.resolve("FooFile.txt")).toFile();
+    // WHEN
+    expectAssertionError(
+        () -> files.assertIsDirectoryRecursivelyContaining(INFO, existingFile, TXT_EXTENSION_PATTERN));
+    // THEN
+    verify(failures).failure(INFO, shouldBeDirectory(existingFile));
+  }
+
+  @Test
+  void should_fail_if_actual_is_empty() {
+    // WHEN
+    expectAssertionError(
+        () -> files.assertIsDirectoryRecursivelyContaining(INFO, tempDirAsFile, TXT_EXTENSION_PATTERN));
+    // THEN
+    verify(failures)
+        .failure(INFO, directoryShouldContain(tempDirAsFile, emptyList(), TXT_EXTENSION_PATTERN_DESCRIPTION));
+  }
+
+  @Test
+  void should_fail_if_actual_does_not_contain_any_files_matching_the_given_pathMatcherPattern() {
+    // GIVEN
+    Path fooDir = createDirectory(tempDir, "foo", "foo2.data");
+    createDirectory(fooDir, "foo3");
+    // WHEN
+    expectAssertionError(
+        () -> files.assertIsDirectoryRecursivelyContaining(INFO, tempDirAsFile, TXT_EXTENSION_PATTERN));
+    // THEN
+    verify(failures)
+        .failure(INFO,
+                 directoryShouldContain(tempDirAsFile, toAbsolutePaths(list(new File(tempDirAsFile, "foo"),
```

Comment:
does `directoryShouldContain` error message is explicit about the fact that the directory was explored recursively?

Reply:
No, but in the failure, the actual directories are listed recursively :
For example : 
`Expecting directory:
  </tmp/junit6913231803498845707>
to contain at least one file matching the 'regex:.+\.txt' pattern but there was none.`
`The directory content was:
  [/tmp/junit6913231803498845707/foo, /tmp/junit6913231803498845707/foo/foo2.data, /tmp/junit6913231803498845707/foo/foo3]
);`

I can replace "Expecting directory..." by "Expecting directory or any of its subdirectories(recursively) ..." if you would like.

Reply:
let's do that in case there was not subdirectories it will make it clear what the assertion tried to do.

Reply:
Not sure to get you there :)

Reply:
sorry I was not clear, your suggestion works well:

> I can replace "Expecting directory..." by "Expecting directory or any of its subdirectories(recursively) ..." if you would like.

Reply:
No problem. Ok I will update consequently.